### PR TITLE
Retries for Electrum requests

### PR DESCRIPTION
**Pull request by nkuba**
We need to retry requests to Electrum endpoints as some of them are failing due to server errors.

Depends on: https://github.com/keep-network/tbtc-v2/pull/516
Closes: https://github.com/keep-network/tbtc-v2/issues/499
Closes: https://github.com/keep-network/tbtc-v2/issues/485

### CODE REVIEW

**Comment by nkuba** [comment]
I make it a draft until https://github.com/keep-network/tbtc-v2/pull/516 is merged.

**Review by tomaszslabon — APPROVED** [review]

**Comment by tomaszslabon** [comment]
It's OK, no comments from me.

File: typescript/src/electrum.ts [hunk 1]
```diff
@@ -51,21 +52,32 @@ type Action<T> = (electrum: any) => Promise<T>
 export class Client implements BitcoinClient {
   private credentials: Credentials
   private options?: ClientOptions
+  private totalRetryAttempts: number
 
-  constructor(credentials: Credentials, options?: ClientOptions) {
+  constructor(
+    credentials: Credentials,
+    options?: ClientOptions,
+    totalRetryAttempts = 3
```
[the remark this answers is not in the document]
Comment (nkuba) [reply typescript/src/electrum.ts:60]: What starting value do you suggest @lukasz-zimnoch ?

[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply typescript/src/electrum.ts:60]: I think we should use 10 seconds. This made sense when I worked on Electrum retries for system tests.

[the remark this answers is not in the document]
Comment (pdyraga) [reply typescript/src/electrum.ts:60]: Are we limiting retries to "read" operations? I think this is a good choice for the API but if this is true, we should clarify that this parameter affects only "reads" and not "writes" (I think Bitcoin tx broadcasting is an exception).

[the remark this answers is not in the document]
Comment (nkuba) [reply typescript/src/electrum.ts:60]: The only write we use in electrum is transaction broadcasting. I don't see a reason why we should limit the retries to writes only.

[the remark this answers is not in the document]
Comment (nkuba) [reply typescript/src/electrum.ts]: 95d5f86ecbb2d90c774b3837fd45f157af24aa55

**Review by lukasz-zimnoch — APPROVED** [review]

**Pull request merged**